channel_output_webhook.py

```diff
@@ -16,22 +16,45 @@ class WebhookOutputChannel(OutputChannel):
 
 
     def do_send_alert(self, input_channel=None, canarydrop=None, **kwargs):
-        try:
-            payload = input_channel.format_webhook_canaryalert(
-                                          canarydrop=canarydrop,
-                                          **kwargs)
-            self.generic_webhook_send(simplejson.dumps(payload), canarydrop)
-        except Exception as e:
-            log.err(e)
+
+        slack = "https://hooks.slack.com"
+
+        if (slack in canarydrop['alert_webhook_url']):
+            try:
+                payload = input_channel.format_webhook_canaryalert(
+                                              canarydrop=canarydrop,
+                                              **kwargs)
+                self.generic_webhook_send(simplejson.dumps(payload), canarydrop)
+            except Exception as e:
+                log.err(e)
+        else:
+            try:
+                payload = input_channel.format_webhook_canaryalert(
+                                              canarydrop=canarydrop,
+                                              **kwargs)
+                self.generic_webhook_send(simplejson.dumps(payload), canarydrop)
+            except Exception as e:
+                log.err(e)
 
     def generic_webhook_send(self, payload=None, canarydrop=None):
-        try:
-            response = requests.post(canarydrop['alert_webhook_url'], payload, headers={'content-type': 'application/json'})
-            response.raise_for_status()
-            log.msg('Webhook sent to {url}'.format(url=canarydrop['alert_webhook_url']))
-            return None
-        except requests.exceptions.RequestException as e:
-            log.err("Failed sending request to webhook {url} with error {error}".format(url=canarydrop['alert_webhook_url'],error=e))
-            return e
-    
-    
+
```

Comment:
Since we have already determined the payload and formatted it correctly, there is no need to change `generic_webhook_send`. The changes can be dropped, and it can be left as is (yay for less code!).

Reply:
fixed